src/iccpd/src/mlacp_sync_prepare.c

```diff
@@ -0,0 +1,622 @@
+        return -1;
+
+    /* Prepare for sync request */
+    memset(buf, 0, max_buf_size);
+
+    icc_hdr = (ICCHdr*)buf;
+    tlv = (mLACPAggPortStateTLV*)&buf[sizeof(ICCHdr)];
+
+    /* ICC header */
+    mlacp_fill_icc_header(csm, icc_hdr, msg_len);
+
+    /* Port State TLV */
+    tlv->icc_parameter.u_bit = 0;
+    tlv->icc_parameter.f_bit = 0;
+    tlv->icc_parameter.type = htons(TLV_T_MLACP_AGGREGATOR_STATE);
+    tlv->icc_parameter.len = htons(sizeof(mLACPAggPortStateTLV) - sizeof(ICCParameter));
+
+    tlv->partner_sys_priority = 0;
+    tlv->partner_key = 0;
+    tlv->agg_id = htons(local_if->po_id);
+    tlv->actor_key = 0;
+    tlv->agg_state = local_if->state;
+
+    return msg_len;
+}
+
+/*****************************************
+* Prprare Sync Purge Port
+*
+* ***************************************/
+int mlacp_prepare_for_Aggport_config(struct CSM* csm,
+                                     char* buf, size_t max_buf_size,
+                                     struct LocalInterface* lif, int purge_flag)
+{
+    ICCHdr* icc_hdr = (ICCHdr*)buf;
+    mLACPAggConfigTLV* tlv = (mLACPAggConfigTLV*)&buf[sizeof(ICCHdr)];
+    size_t msg_len = sizeof(ICCHdr) + sizeof(mLACPAggConfigTLV);
+
+    if (csm == NULL)
+        return -1;
+
+    if (buf == NULL)
+        return -1;
+
+    if (msg_len > max_buf_size)
+        return -1;
+
+    /* Prepare for sync request */
+    memset(buf, 0, max_buf_size);
+
+    icc_hdr = (ICCHdr*)buf;
+    tlv = (mLACPAggConfigTLV*)&buf[sizeof(ICCHdr)];
+
+    /* ICC header */
+    mlacp_fill_icc_header(csm, icc_hdr, msg_len);
+
+    /* Port Config TLV */
+    tlv->icc_parameter.u_bit = 0;
+    tlv->icc_parameter.f_bit = 0;
+    tlv->icc_parameter.type = htons(TLV_T_MLACP_AGGREGATOR_CONFIG);
+
+    tlv->icc_parameter.len = htons(sizeof(mLACPAggConfigTLV) - sizeof(ICCParameter));
+    tlv->agg_id = htons(lif->po_id);
+    if (purge_flag == 1)
+        tlv->flags = 0x02; /*purge*/
+    else
+        tlv->flags = 0x1;
+    tlv->agg_name_len = strlen(lif->name);
+    memcpy(tlv->agg_name, lif->name, MAX_L_PORT_NAME);
+    memcpy(tlv->mac_addr, lif->mac_addr, ETHER_ADDR_LEN);
+
+    return msg_len;
+}
+
+/*****************************************
+* Preprare Sync MAC-Info TLV
+*
+* ***************************************/
+int mlacp_prepare_for_mac_info_to_peer(struct CSM* csm, char* buf, size_t max_buf_size, struct MACMsg* mac_msg)
+{
+    struct mLACPMACInfoTLV* tlv = NULL;
+    size_t msg_len = 0;
+    size_t tlv_len = 0;
+    ICCHdr* icc_hdr = NULL;
+
+    if (!csm)
+        return -1;
+    if (!buf)
+        return -1;
+
+    tlv_len = sizeof(struct mLACPMACInfoTLV);
+
+    if ((msg_len = sizeof(ICCHdr) + tlv_len) > max_buf_size)
+        return -1;
+
+    /* ICC header */
+    memset(buf, 0, max_buf_size);
+    icc_hdr = (ICCHdr*)buf;
+    mlacp_fill_icc_header(csm, icc_hdr, msg_len);
+
+    /* Prepare for ARP information TLV */
+    tlv = (struct mLACPMACInfoTLV*)malloc(tlv_len);
```

Comment:
Why do we need to allocate memory for tlv structure? Stack variable can be used to avoid unneeded dynamic memory allocation to minimize memory fragmentation.

Reply:
OK, we will refine this ASAP.